diag_manager/include/fms_diag_fieldbuff_update.fh

```diff
@@ -0,0 +1,1336 @@
+
+      INTEGER, INTENT(inout) :: field_num_threads
+      INTEGER, INTENT(inout) :: field_active_omp_level
+
+      LOGICAL, INTENT(inout) :: issued_mask_ignore_warning
+      INTEGER, DIMENSION(3), INTENT(in)  :: l_start !< local start indices on 3 axes for regional output
+      INTEGER, DIMENSION(3), INTENT(in)  :: l_end !< local end indices on 3 axes for regional output
+      CHARACTER(len=*), INTENT(inout), OPTIONAL :: err_msg
+      CHARACTER(len=256), INTENT(inout) :: err_msg_local
+
+      INTEGER :: pow_value !< A copy of same variable in ofield_cfg
+      CHARACTER(:), ALLOCATABLE :: output_name !< A copy of same variable in ofield_cfg
+      CHARACTER(:), ALLOCATABLE :: field_name  !< A copy of same variable in ofield_cfg
+      CHARACTER(:), ALLOCATABLE :: module_name !< A copy of same variable in ofield_cfg
+      LOGICAL :: phys_window !< A copy of same variable in ofield_cfg
+      LOGICAL :: need_compute !< A copy of same variable in ofield_cfg
+      LOGICAL :: reduced_k_range !< A copy of same variable in ofield_cfg
+      LOGICAL :: mask_variant !< A copy of same variable in ofield_cfg
+      LOGICAL :: mask_present !< A copy of same variable in ofield_cfg
+      LOGICAL :: missvalue_present !< A copy of same variable in ofield_cfg
+
+      !> The indecies copied directly from the ofield_index_cfg:
+      INTEGER:: is, js, ks, ie, je, ke, hi, hj, f1, f2, f3, f4
+      INTEGER:: ls, le !> start and end indecies for the 4th dimension.
+
+      INTEGER :: ksr, ker !> Loop indeccies used in reduced_k_range calculations
+      !> Looping indecies, derived from ofield_index_cfg:
+      INTEGER :: i, j, k, l, i1, j1, k1
+
+      INTEGER :: numthreads
+      INTEGER :: active_omp_level
+
+      LOGICAL :: succeded !> True iff no errors encountered.
+      CHARACTER(len=128):: error_string
+
+#if defined(_OPENMP)
+      INTEGER :: omp_get_num_threads !< OMP function
+      INTEGER :: omp_get_level !< OMP function
+#endif
+
+      !!TODO: (MDM) Will the interface allow passing in is, ie?
+      ls = 1
+      le = SIZE(field_d, 4)
+
+      ksr= l_start(3)
+      ker= l_end(3)
+      is = ofield_index_cfg%is
+      js = ofield_index_cfg%js
+      ks = ofield_index_cfg%ks
+      ie = ofield_index_cfg%ie
+      je = ofield_index_cfg%je
+      ke = ofield_index_cfg%ke
+      hi = ofield_index_cfg%hi
+      hj = ofield_index_cfg%hj
+      f1 = ofield_index_cfg%f1
+      f2 = ofield_index_cfg%f2
+      f3 = ofield_index_cfg%f3
+      f4 = ofield_index_cfg%f4
+
+      output_name = trim(ofield_cfg%output_name)
+      field_name = trim(ofield_cfg%field_name)
+      module_name = trim(ofield_cfg%module_name)
+      pow_value = ofield_cfg%pow_value
+      phys_window = ofield_cfg%phys_window
+      reduced_k_range = ofield_cfg%reduced_k_range
+      need_compute = ofield_cfg%need_compute
+      mask_variant =  ofield_cfg%mask_variant
+      mask_present =  ofield_cfg%mask_present
+      missvalue_present = ofield_cfg%missvalue_present
```

Comment:
There should be getter functions for all of these variables because they should all be private.